code/datums/components/keypad_lock.dm

```diff
@@ -16,6 +16,18 @@
 
 	//If the panel is open
 	var/panel_open = FALSE
+	//Last viable keypad_input
+	var/memory_code = ""
+	//If it is forced
+	var/is_forced = FALSE
+	//Strength of lock: determines how easy/hard it is to bypass lock
+	//0 = force open and hacking | 1 = hacking only | 2 = hacking only and no reseting memory
+	var/lock_strength = 0
+	//List of items that can force keypad locks
+	var/list/keypad_tools = list(/obj/item/jawsoflife, 
```

Comment:
Might be worth adding the light eater to this if the mantis blade and armblade already are